azurerm/internal/services/policyinsights/tests/resource_arm_policy_remediation_test.go

```diff
@@ -0,0 +1,591 @@
+				),
+			},
+			{
+				Config: testAccAzureRMPolicyRemediation_updateLocation(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyRemediationExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "location_filters.#", "1"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMPolicyRemediation_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_policy_remediation", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPolicyRemediationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPolicyRemediation_atResourceGroup(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyRemediationExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMPolicyRemediation_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMPolicyRemediationExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy Insights Remediation not found: %s", resourceName)
+		}
+
+		id, err := parse.RemediationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).PolicyInsights.RemediationsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := policyinsights.RemediationGetAtScope(ctx, client, id.Name, id.RemediationScopeId); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Policy Insights Remediation %q (Scope %q) does not exist", id.Name, id.ScopeId)
+			}
+			return fmt.Errorf("Bad: Get on PolicyInsights.RemediationsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPolicyRemediationDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).PolicyInsights.RemediationsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_policy_remediation" {
+			continue
+		}
+
+		id, err := parse.RemediationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := policyinsights.RemediationGetAtScope(ctx, client, id.Name, id.RemediationScopeId); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on remediationsClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPolicyRemediation_atSubscription(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+data "azurerm_subscription" "current" {}
```

Comment:
We'll need to add a provider block to all these tests

Reply:
Done